[2025-01-24T19:53:17.013+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-01-24T19:53:17.029+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_clickhouse_test.connect_to_clickhouse manual__2025-01-24T19:53:13.744990+00:00 [queued]>
[2025-01-24T19:53:17.038+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_clickhouse_test.connect_to_clickhouse manual__2025-01-24T19:53:13.744990+00:00 [queued]>
[2025-01-24T19:53:17.038+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2025-01-24T19:53:17.051+0000] {taskinstance.py:2889} INFO - Executing <Task(_PythonDecoratedOperator): connect_to_clickhouse> on 2025-01-24 19:53:13.744990+00:00
[2025-01-24T19:53:17.058+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=846) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-01-24T19:53:17.059+0000] {standard_task_runner.py:72} INFO - Started process 848 to run task
[2025-01-24T19:53:17.060+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'spark_clickhouse_test', 'connect_to_clickhouse', 'manual__2025-01-24T19:53:13.744990+00:00', '--job-id', '107', '--raw', '--subdir', 'DAGS_FOLDER/spark_clickhouse_test.py', '--cfg-path', '/tmp/tmp1j4xbfgo']
[2025-01-24T19:53:17.062+0000] {standard_task_runner.py:105} INFO - Job 107: Subtask connect_to_clickhouse
[2025-01-24T19:53:17.101+0000] {task_command.py:467} INFO - Running <TaskInstance: spark_clickhouse_test.connect_to_clickhouse manual__2025-01-24T19:53:13.744990+00:00 [running]> on host 39a8c5c570e1
[2025-01-24T19:53:17.186+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Smalch' AIRFLOW_CTX_DAG_ID='spark_clickhouse_test' AIRFLOW_CTX_TASK_ID='connect_to_clickhouse' AIRFLOW_CTX_EXECUTION_DATE='2025-01-24T19:53:13.744990+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-01-24T19:53:13.744990+00:00'
[2025-01-24T19:53:17.187+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-01-24T19:53:17.191+0000] {connection.py:411} WARNING - Failed to connect to None:9000
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/connection.py", line 399, in connect
    return self._init_connection(host, port)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/connection.py", line 329, in _init_connection
    self.socket = self._create_socket(host, port)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/connection.py", line 301, in _create_socket
    raise err
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/connection.py", line 292, in _create_socket
    sock.connect(sa)
OSError: [Errno 99] Cannot assign requested address
[2025-01-24T19:53:17.193+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/decorators/base.py", line 266, in execute
    return_value = super().execute(context)
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/spark_clickhouse_test.py", line 56, in connect_to_clickhouse
    result = client.execute(query)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 370, in execute
    with self.disconnect_on_error(query, settings):
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 313, in disconnect_on_error
    self.establish_connection(settings)
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 300, in establish_connection
    self.connection.force_connect()
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/connection.py", line 258, in force_connect
    self.connect()
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/connection.py", line 420, in connect
    raise err
clickhouse_driver.errors.NetworkError: Code: 210. Cannot assign requested address (None:9000)
[2025-01-24T19:53:17.205+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=spark_clickhouse_test, task_id=connect_to_clickhouse, run_id=manual__2025-01-24T19:53:13.744990+00:00, execution_date=20250124T195313, start_date=20250124T195317, end_date=20250124T195317
[2025-01-24T19:53:17.217+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-01-24T19:53:17.218+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 107 for task connect_to_clickhouse (Code: 210. Cannot assign requested address (None:9000); 848)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/decorators/base.py", line 266, in execute
    return_value = super().execute(context)
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/spark_clickhouse_test.py", line 56, in connect_to_clickhouse
    result = client.execute(query)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 370, in execute
    with self.disconnect_on_error(query, settings):
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 313, in disconnect_on_error
    self.establish_connection(settings)
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/client.py", line 300, in establish_connection
    self.connection.force_connect()
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/connection.py", line 258, in force_connect
    self.connect()
  File "/home/airflow/.local/lib/python3.12/site-packages/clickhouse_driver/connection.py", line 420, in connect
    raise err
clickhouse_driver.errors.NetworkError: Code: 210. Cannot assign requested address (None:9000)
[2025-01-24T19:53:17.233+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-01-24T19:53:17.256+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-01-24T19:53:17.260+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
